### ISSUE TITLE
Continued Training: Refusing to overwrite model folder

### ISSUE THREAD

**Issue by LiangQiqi677** [awslabs/sockeye#934]
Hi, I have a model training that is interrupted. I want to resume training.

when I relaunched it, it says it refuse to overwrite model folder as it seems to contain a trained model. 
The whole training script is unchanged. I didnt change the file, I just run it again but I got this error.

May you share how to fix this error? Thanks!

**Comment by LiangQiqi677** [comment]
I check the code, it seems that I don't have the training_state_dir. Can you show what is the training_state_dir? And I didnt know where to add the "training_state_dir" param into my code. My output_dir dodn't contain the training_state?


`def check_resume(args: argparse.Namespace, output_folder: str) -> bool:
    """
    Check if we should resume a broken training run.
    :param args: Arguments as returned by argparse.
    :param output_folder: Main output folder for the model.
    :return: Flag signaling if we are resuming training and the directory with
        the training status.
    """
    resume_training = False
    training_state_dir = os.path.join(output_folder, C.TRAINING_STATE_DIRNAME)
    if os.path.exists(output_folder):
        if args.overwrite_output:
            logger.info("Removing existing output folder %s.", output_folder)
            shutil.rmtree(output_folder)
            os.makedirs(output_folder)
        elif os.path.exists(training_state_dir):
            old_args = vars(arguments.load_args(os.path.join(output_folder, C.ARGS_STATE_NAME)))
            arg_diffs = _dict_difference(vars(args), old_args) | _dict_difference(old_args, vars(args))
            # Remove args that may differ without affecting the training.
            arg_diffs -= set(C.ARGS_MAY_DIFFER)
            # allow different device-ids provided their total count is the same
            if 'device_ids' in arg_diffs and len(old_args['device_ids']) == len(vars(args)['device_ids']):
                arg_diffs.discard('device_ids')
            if not arg_diffs:
                resume_training = True
            else:
                # We do not have the logger yet
                logger.error("Mismatch in arguments for training continuation.")
                logger.error("Differing arguments: %s.", ", ".join(arg_diffs))
                sys.exit(1)
        elif os.path.exists(os.path.join(output_folder, C.PARAMS_BEST_NAME)):
            logger.error("Refusing to overwrite model folder %s as it seems to contain a trained model.", output_folder)
            sys.exit(1)`


And this is my output_dir:
-rw-rw-r-- 1 liangqiqi liangqiqi 5.0K 3月   7 23:35 args.yaml
-rw-rw-r-- 1 liangqiqi liangqiqi 6.7K 3月   7 23:37 config
-rw-rw-r-- 1 liangqiqi liangqiqi  493 3月   7 23:37 data.info
-rw-rw-r-- 1 liangqiqi liangqiqi 479K 3月   8 10:36 log
-rw-rw-r-- 1 liangqiqi liangqiqi  15K 3月   8 10:36 metrics
-rw-rw-r-- 1 liangqiqi liangqiqi 362M 3月   8 10:36 params.00000
...
-rw-rw-r-- 1 liangqiqi liangqiqi 362M 3月   8 10:36 params.00063
lrwxrwxrwx 1 liangqiqi liangqiqi   12 3月   8 01:53 params.best -> params.00013
-rw-rw-r-- 1 liangqiqi liangqiqi 175K 3月   7 23:37 symbol.json
-rw-rw-r-- 1 liangqiqi liangqiqi    7 3月   7 23:37 version
-rw-rw-r-- 1 liangqiqi liangqiqi 313K 3月   7 23:37 vocab.src.0.json
-rw-rw-r-- 1 liangqiqi liangqiqi 458K 3月   7 23:37 vocab.trg.0.json

**Comment by bricksdont** [comment]
Hi, Please show your entire training command. Thanks!

**Comment by LiangQiqi677** [comment]
python -m sockeye.train -source ${TRAINING_SOURCE} \
                        -target ${TRAINING_TARGET} \
                        -validation-source ${VALIDATION_SOURCE} \
                        -validation-target ${VALIDATION_TARGET} \
                        --batch-size 2048 \
                        --checkpoint-interval 2000 \
                        --learning-rate-reduce-num-not-improved 3 \
                        --max-num-checkpoint-not-improved 4 \
                        --seed 100 \
                        --decode-and-evaluate 0 \
                        --output ${MODEL_DIR_OUTPUT} \
                        --device-ids 6 7


> Hi, Please show your entire training command. Thanks!

**Comment by bricksdont** [comment]
What do you mean by your model training was "interrupted"? Please also attach your entire training log.

**Comment by LiangQiqi677** [comment]
Maybe my statement is inaccurate. 

The first one situation：
Validation-perplexity has not improved for 4 checkpoints, so the training finished automatically. 
At this time can I use one of my checkpoints(e.g. params.best) to resume the training?

The second situation：
Sometimes I kill the program by mistake, and I want to resume from the lastest ckpt.

**Comment by bricksdont** [comment]
For both situations, please attach a training log file.

Without having seen the logs:

first situation: if `--max-num-checkpoint-not-improved` is reached, Sockeye will refuse to train more. Increase `--max-num-checkpoint-not-improved` to something higher than 4. (There might be other arguments you need to adjust as well.)

second situation: if the training did not finish, Sockeye will resume training and not produce the error you show above. You would need to show that this is happening to you with scripts that someone else can use to reproduce the problem.

**Comment by LiangQiqi677** [comment]
Thx!
first situation: maybe next time I should try to set --max-num-checkpoint-not-improved higher
second situation: I did resume training
